recipes/icecream-cpp/all/conanfile.py

```diff
@@ -0,0 +1,15 @@
+from conans import ConanFile, tools
+
+class IcecreamcppConan(ConanFile):
+    name = "icecream-cpp"
+    license = "MIT"
+    url = "https://github.com/renatoGarcia/icecream-cpp"
+    description = "A little library to help with the print debugging on C++11 and forward."
+    topics = ("debug", "single-header-lib", "print")
+    no_copy_source = True
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
```

Comment:
If you rename the source folder once here, you don't need to use version dependent variables in package.

Reply:
I'm sorry. I don't get what you mean by that.

Reply:
```suggestion
        tools.get(**self.conan_data["sources"][self.version])
        os.rename("icecream-cpp-{}".format(self.version), self._source_subfolder)
```
And add a `_source_subfolder = "source_subfoler"` member variable.
You can then use this variable in the `package` method.
That way, this recipe has the same format as other.

Reply:
Is that really needed? With your suggested changes I will be importing a new module (`os`), making a call to one more method (`os.rename`) and defining a new member variable (`_source_subfolder`). IMHO with that I will be adding more complexity to a code working just fine as it is, to no obvious gain.

Reply:
Import in python is negligible. `conan` overhead is much bigger and uses it itself.
Your packages will have the same expected format as all others, saving time in the future.

Reply:
I was not arguing about performance, but code complexity like in "understanding the code" complexity. But whatever, this code is so small, and if you say that is the common practice. I have made the changes.

Reply:
Thanks!
There is also complexity in diversity of coding styles. :smile:

Reply:
I concede that point. :smile: